### PR TITLE
PIC-994 disregard messages for courts which are not configured

### PR DESCRIPTION
**Pull request by chris-faulkner-mojdt**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/xml/DocumentUtils.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    private const val DELIMITER = "_"
+
+    fun getCourtCode(documents: Element): String? {
+        // XPathFactory not thread safe so make one each time
+        val xPath: XPath = XPathFactory.newInstance().newXPath()
+        val exp = xPath.compile(SOURCE_FILE_NAME_EXPR)
+        val nodeList = exp.evaluate(documents, XPathConstants.NODESET) as NodeList
+        for (i in 0 until nodeList.length) {
+            getOuCode(nodeList.item(i))?.let {
+                code ->
```
Comment (chris-faulkner-mojdt) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/xml/DocumentUtils.kt:25]: The dps-gradle-spring-boot seems to require this strange format in its linting checks

File: src/main/resources/xsd/cp/external/ExternalDocumentRequest.xsd [hunk 2]
```diff
@@ -27,7 +27,7 @@
     <xs:element name="ExternalDocumentRequest">
         <xs:complexType>
             <xs:sequence>
-                <xs:element name="documents" type="tns:documents" minOccurs="0" maxOccurs="1"/>
+                <xs:element name="documents" type="tns:documents" minOccurs="1" maxOccurs="1"/>
```
Comment (chris-faulkner-mojdt) [inline src/main/resources/xsd/cp/external/ExternalDocumentRequest.xsd:30]: This is in line with what we find in the CGI XSD

File: src/test/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/integration/endpoint/ExternalDocRequestEndpointIntTest.kt [hunk 3]
```diff
@@ -70,16 +72,62 @@ class ExternalDocRequestEndpointIntTest : IntegrationTestBase() {
...
+    fun `should not enqueue message when court is not processed but return acknowledgement`() {
+        val externalDoc1 = readFile("src/test/resources/soap/ignored-courts.xml")
+        val requestEnvelope: Source = StringSource(externalDoc1)
 
         mockClient.sendRequest(RequestCreators.withSoapEnvelope(requestEnvelope))
-            .andExpect(serverOrReceiverFault())
+            .andExpect(validPayload(xsdResource))
+            .andExpect(
+                xpath("//ns3:Acknowledgement/ackType/MessageComment", namespaces)
+                    .evaluatesTo("Message ignored - the court B10XX in the message is not processed")
```
Comment (jhevans) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/integration/endpoint/ExternalDocRequestEndpointIntTest.kt:84]: Do we know that CGI will ignore the message returned in the Ack? One would imagine so and definitely nice to be explicit about what's happening but might be worth checking if we're not sure.

**Review by jhevans — APPROVED** [review]
LGTM 👍

**Pull request merged**